agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -143,6 +146,19 @@ func (state *DockerTaskEngineState) ContainerMapByArn(arn string) (map[string]*a
 	return ret, ok
 }
 
+// TaskByShortID retrieves the task of a given docker short container id
+func (state *DockerTaskEngineState) TaskByShortID(cid string) (*api.Task, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	for id := range state.idToTask {
+		if strings.HasPrefix(id, cid) {
```

Comment:
Yeah, I think I'd prefer returning an error when collisions are found rather than randomly returning one of the matching tasks.